grails-app/controllers/org/pih/warehouse/api/CycleCountApiController.groovy

```diff
@@ -45,21 +48,98 @@ class CycleCountApiController {
         BatchCommandUtils.validateBatch(command, "requests")
         List<CycleCountDto> cycleCounts = cycleCountService.startCycleCount(command)
 
-        render([data: cycleCounts] as JSON)
+        if (!params.format) {
+            params.format = "json"
+        }
+
+        withFormat {
+            xls {
+                exportCountXls(cycleCounts)
+            }
+            pdf {
+                renderCountPdf(cycleCounts, command.facility.name)
+            }
+            json {
+                render([data: cycleCounts] as JSON)
+            }
+        }
     }
 
     def startRecount(CycleCountStartRecountBatchCommand command) {
         BatchCommandUtils.validateBatch(command, "requests")
         List<CycleCountDto> cycleCounts = cycleCountService.startRecount(command)
 
-        render([data: cycleCounts] as JSON)
+        if (!params.format) {
+            params.format = "json"
+        }
+
+        withFormat {
+            xls {
+                exportRecountXls(cycleCounts)
+            }
+            pdf {
+                renderRecountPdf(cycleCounts, command.facility.name)
+            }
+            json {
+                render([data: cycleCounts] as JSON)
+            }
+        }
     }
 
     def list() {
         List<String> ids = params.list("id")
         List<CycleCountDto> cycleCounts = cycleCountService.getCycleCounts(ids)
 
-        render([data: cycleCounts] as JSON)
+        if (!params.format) {
+            params.format = "json"
+        }
+
+        boolean isRecount = cycleCounts?.any { it.status == CycleCountStatus.INVESTIGATING.name() }
+
+        withFormat {
+            xls {
+                isRecount ? exportRecountXls(cycleCounts) : exportCountXls(cycleCounts)
+            }
+            pdf {
+                String facilityName = cycleCounts?.first()?.cycleCountItems?.first()?.facility?.name  ?: ""
```

Comment:
this is fine, but it might be easier to get the facility from the URI

Reply:
I did not realize it was there. I thought this was a simple cycle count get endpoint without facility mixed in because the facility id is not even used there. To be fair you, could provide a different facility id and this endpoint still would work, so I prefer pulling it out from the actual cycle count.

Reply:
yeah that's fair. We haven't really discussed in depth the implications of prefixing all our APIs with `/api/facilities/$facilityId/`. To me if we're requiring it in the URI we should be using it in validation (and if we're not, we shouldn't have it in the uri). The idea is probably that we don't want people to be able to fetch data from other facilities that they shouldn't have access to.

But again we haven't really discussed this so I agree with you like this is fine for now.